### PR TITLE
Quick merge of Rohan's fork

### PR DESCRIPTION
**Pull request by jmgoldb**
I did a quick merge of the fork Rohan made that added [NSManagedObject vok_fetchAllForPredicate:sortedByKey:ascending:forManagedObjectContext]
and [NSManagedObject vok_fetchAllForPredicate:sortedBy:forManagedObjectContext] as well as [VOKCoreDataManager arrayForClass:withPredicate:sortedBy:forContext]  with the current master branch of Vokoder.

I've ran all of the existing Unit tests on the sample project and everything still passes just fine, and the code itself looks fine as well.  Is there other work that I can be doing to help this along, or is this ready for a review?

### CODE REVIEW

File: Pod/Classes/NSManagedObject+VOKManagedObjectAdditions.h [hunk 1]
```diff
@@ -97,6 +97,30 @@
...
 
 /*
+ Returns all entites matching the predicate.
+ @param predicate       Predicate to use to fetch.
+ @param sortDescriptors SortDescriptors to use to sort the results.
+ @param contextOrNil    The managed object context to fetch in.  If nil, the main context will be used.
+ @return                NSArray full of the instances of the current class.
+ */
++ (NSArray *)vok_fetchAllForPredicate:(NSPredicate *)predicate
+                             sortedBy:(NSArray*)sortDescriptors
```
Comment (seanwolter) [inline Pod/Classes/NSManagedObject+VOKManagedObjectAdditions.h]: missing space before the asterisk

File: Pod/Classes/NSManagedObject+VOKManagedObjectAdditions.m [hunk 2]
```diff
@@ -124,6 +124,30 @@ + (NSArray *)vok_fetchAllForPredicate:(NSPredicate *)predicate forManagedObjectC
                                                    forContext:contextOrNil];
 }
 
++ (NSArray *)vok_fetchAllForPredicate:(NSPredicate *)predicate
+                             sortedBy:(NSArray*)sortDescriptors
```
Comment (seanwolter) [inline Pod/Classes/NSManagedObject+VOKManagedObjectAdditions.m]: missing space before the asterisk

File: Pod/Classes/NSManagedObject+VOKManagedObjectAdditions.h [hunk 3]
```diff
@@ -97,6 +97,30 @@
 + (NSArray *)vok_fetchAllForPredicate:(NSPredicate *)predicate forManagedObjectContext:(NSManagedObjectContext *)contextOrNil;
 
 /*
+ Returns all entites matching the predicate.
```
Comment (seanwolter) [inline Pod/Classes/NSManagedObject+VOKManagedObjectAdditions.h]: this docu-comment should be updated to mention the sorting

File: Pod/Classes/NSManagedObject+VOKManagedObjectAdditions.h [hunk 4]
```diff
@@ -97,6 +97,30 @@
...
+ @param sortDescriptors SortDescriptors to use to sort the results.
+ @param contextOrNil    The managed object context to fetch in.  If nil, the main context will be used.
+ @return                NSArray full of the instances of the current class.
+ */
++ (NSArray *)vok_fetchAllForPredicate:(NSPredicate *)predicate
+                             sortedBy:(NSArray *)sortDescriptors
+              forManagedObjectContext:(NSManagedObjectContext *)contextOrNil;
+
+/*
+ Returns all entites matching the predicate.
```
Comment (seanwolter) [inline Pod/Classes/NSManagedObject+VOKManagedObjectAdditions.h]: this docu-comment should also be updated to mention the sorting + ascending option

File: Pod/Classes/NSManagedObject+VOKManagedObjectAdditions.h [hunk 5]
```diff
@@ -97,6 +97,30 @@
...
+ @param predicate       Predicate to use to fetch.
+ @param sortKey         Key to use to sort the results.
+ @param ascending       Whether not to sort in ascending or descending.
```
Comment (seanwolter) [inline Pod/Classes/NSManagedObject+VOKManagedObjectAdditions.h]: `Whether not` missing an "or"

File: Pod/Classes/NSManagedObject+VOKManagedObjectAdditions.h [hunk 6]
```diff
@@ -104,4 +128,6 @@
  */
 + (instancetype)vok_fetchForPredicate:(NSPredicate *)predicate forManagedObjectContext:(NSManagedObjectContext *)contextOrNil;
 
+
```
Comment (seanwolter) [inline Pod/Classes/NSManagedObject+VOKManagedObjectAdditions.h]: looks like extra whitespace

File: Pod/Classes/VOKCoreDataManager.m [hunk 7]
```diff
@@ -327,12 +327,30 @@ - (NSArray *)arrayForClass:(Class)managedObjectClass withPredicate:(NSPredicate
...
+{
+    contextOrNil = [self safeContext:contextOrNil];
+    NSFetchRequest *fetchRequest = [self fetchRequestWithClass:managedObjectClass
+                                                     predicate:predicate
+                                               sortDescriptors:sortDescriptors];
+    return [self arrayForFetchRequest:fetchRequest inContext:contextOrNil];
+}
+
+-(NSArray *)arrayForFetchRequest:(NSFetchRequest *)fetchRequest
+                      inContext:(NSManagedObjectContex t*)context
```
Comment (seanwolter) [inline Pod/Classes/VOKCoreDataManager.m]: misaligned colons :scream:

Comment (chillpop) [inline Pod/Classes/VOKCoreDataManager.m]: and what's that extra 't' doing there?

File: SampleProject/VOKCoreDataManagerTests/VOKManagedObjectAdditions.m [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+- (void)tearDown
+{
+    [super tearDown];
+    [[VOKCoreDataManager sharedInstance] deleteAllObjectsOfClass:[VIThing class] context:nil];
+    [[VOKCoreDataManager sharedInstance] saveAndMergeWithMainContext:nil];
+}
+
+#pragma mark - Test Data Helper Methods
+
+#define basicTestDataStartPoint 0
```
Comment (seanwolter) [inline SampleProject/VOKCoreDataManagerTests/VOKManagedObjectAdditions.m]: these should be `static const` NSIntegers or NSUIntegers

File: SampleProject/VOKCoreDataManagerTests/VOKManagedObjectAdditions.m [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+//
+
+#import <UIKit/UIKit.h>
+#import <XCTest/XCTest.h>
+
+#import <VOKCoreDataManager.h>
+#import <NSManagedObject+VOKManagedObjectAdditions.h>
+#import "VIThing.h"
+
+
```
Comment (seanwolter) [inline SampleProject/VOKCoreDataManagerTests/VOKManagedObjectAdditions.m]: extra line of whitespace here I believe

**Comment by seanwolter** [comment]
@rpanchal 's hard work addresses the oldest issue from this repo
https://github.com/vokal/Vokoder/issues/1

**Comment by seanwolter** [comment]
This also addresses https://github.com/vokal/Vokoder/issues/31 "Add sorted fetch methods in category"

**Comment by seanwolter** [comment]
Could you please pull from upstream and push to your repo once more? (to make sure Brock's changes are a-ok mixed with these)

**Comment by jmgoldb** [comment]
Will do

**Comment by jmgoldb** [comment]
Alright, let's see what Travis says now.

**Comment by seanwolter** [comment]
Travis.org keeps timing out, even though test pass locally in no time.

**Pull request merged**